zeapp/android/src/main/java/de/berlindroid/zeapp/zerepositories/ZeSlotRepository.kt

```diff
@@ -0,0 +1,309 @@
+            zePreferencesService.dataStore.edit { mutablePreferences ->
+                // Getting all related information about the slot config.
+                val slotRelatedKeys =
+                    mutablePreferences.asMap().keys.filter {
+                        it.name.contains(slot.preferencesKeyPrefix())
+                    }
+
+                // Deleting each key that refers to that slot config.
+                slotRelatedKeys.forEach { mutablePreferences.remove(it) }
+            }
+        }
+
+        fun getDefaultSlotConfiguration(slot: ZeSlot): ZeConfiguration =
+            when (slot) {
+                is ZeSlot.Name ->
+                    ZeConfiguration.Name(
+                        null,
+                        null,
+                        imageProviderService.getInitialNameBitmap(),
+                    )
+
+                is ZeSlot.FirstSponsor -> ZeConfiguration.Picture(R.drawable.page_google.toBitmap())
+                is ZeSlot.FirstCustom -> ZeConfiguration.Picture(R.drawable.soon.toBitmap())
+                is ZeSlot.SecondCustom -> ZeConfiguration.Picture(R.drawable.soon.toBitmap())
+                ZeSlot.QRCode ->
+                    ZeConfiguration.QRCode(
+                        title = "",
+                        text = "",
+                        url = "",
+                        isVcard = false,
+                        phone = "",
+                        email = "",
+                        bitmap = R.drawable.qrpage_preview.toBitmap(),
+                    )
+
+                ZeSlot.Weather ->
+                    ZeConfiguration.Weather(
+                        LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE),
+                        "22C",
+                        R.drawable.soon.toBitmap(),
+                    )
+
+                is ZeSlot.Quote ->
+                    ZeConfiguration.Quote(
+                        "Test",
+                        "Author",
+                        R.drawable.page_quote_sample.toBitmap(),
+                    )
+
+                ZeSlot.BarCode ->
+                    ZeConfiguration.BarCode(
+                        "Your title for barcode",
+                        "",
+                        R.drawable.soon.toBitmap(),
+                    )
+
+                ZeSlot.Add ->
+                    ZeConfiguration.Name(
+                        null,
+                        null,
+                        imageProviderService.provideImageBitmap(R.drawable.add),
+                    )
+
+                ZeSlot.Camera ->
+                    ZeConfiguration.Camera(
+                        imageProviderService.provideImageBitmap(R.drawable.soon),
+                    )
+            }
+
+        fun getInitialSlots(): List<ZeSlot> =
+            listOf(
+                ZeSlot.Name,
+                ZeSlot.FirstSponsor,
+                ZeSlot.Camera,
+                ZeSlot.Add,
+            )
```

Comment:
Not sure if this is the best place for this method to be, but other than that, this will allow us to decouple other functionalities that were attached to the `ZeBadgeViewModel`, for example, now the **simulator** can access this repository, and have its own `ViewModel`.

Reply:
I like that, at some point we should also think about decoupling the name from the configuration and make the add configuration (????) actually a thing not a configuration.... Anyway, good approach really like the separation.

Reply:
Yeah, definitelly! I'm still struggling a bit to understand that a slot refers to a configuration and not "have" a configuration as an attribute on a single entity.
But agree with future decoupling work.